gradio/components/gallery.py

```diff
@@ -155,13 +157,20 @@ def postprocess(
                 file_path = str(utils.abspath(file))
             elif isinstance(img, str):
                 file_path = img
-                url = img if is_http_url_like(img) else None
+                if is_http_url_like(img):
+                    url = img
+                    orig_name = Path(img).name
+                else:
+                    url = None
+                    orig_name = Path(urlparse(img).path).name
```

Comment:
Shouldn't the `urlparse` be happening if `is_http_url_like` ?

Reply:
my bad 😣
Fixed -> [f7978e8](https://github.com/gradio-app/gradio/pull/6487/commits/f7978e8cff83472dfe634d3313133732e623dba9)